### DISCUSSION TITLE
Support distributed key

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#8277] [Ideas]
Since databend is using cloud service as its storage infra, it'll be easier to have hash-based strategy partition.
```
create table ... Engine = Fuse
DISTRIBUTED BY HASH(k1)
max_hash_bucket = 32
```

This may help in performance. Such as hash join and aggregate(distinct),  which may avoid local shuffle and benefit co-located join.

**Comment by xudong963** [comment]
Agree

**Reply by wubx** [reply]
Agree. The distributed key is a great performance improvement.

**Comment by sundy-li** [comment]
Refer to:

1. https://www.sisense.com/blog/double-your-redshift-performance-with-the-right-sortkeys-and-distkeys/
2. https://doris.apache.org/zh-CN/docs/advanced/join-optimization/doris-join-optimization/

**Reply by artorias1024** [reply]
Hi,Is there a corresponding pull request or RFC to track this idea?

**Reply by sundy-li** [reply]
Not yet, still investigating.
